standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -5322,16 +5380,25 @@ private void copyMSD(MStorageDescriptor newSd, MStorageDescriptor oldSd) {
         oldSd.setCD(newSd.getCD());
       }
     }
+    // If the serde info of the old SD != the serde info of the new one,
+    // then we set the serde info of the old SD as the new serde info.
+    SerDeInfo oldSerDeInfo = null;
+    SerDeInfo newSerDeInfo = null;
+    try {
+      oldSerDeInfo = convertToSerDeInfo(oldSd.getSerDeInfo(), true);
+      newSerDeInfo = convertToSerDeInfo(newSd.getSerDeInfo(), true);
+    } catch (MetaException e) {
+      LOG.debug("convertToSerDeInfo shouldn't throw MetaException when allowNull is set to true");
+    }
+    if (oldSerDeInfo == null || !oldSerDeInfo.equals(newSerDeInfo)) {
+      oldSd.setSerDeInfo(newSd.getSerDeInfo());
```

Comment:
can we reuse the variable from L5389?

Reply:
They are different types, SerDeInfo and MSerDeInfo. `newSd.getSerDeInfo()` actually returns MSerDeInfo.